### ISSUE TITLE
Issues with either messages not going through or receiving confirmation from the receiver

### ISSUE THREAD

**Issue by tomsmith8** [stakwork/sphinx-relay#778]
As per this ticket [here](https://github.com/stakwork/sphinx-ios/issues/309) some messages are not going through or there is an issue with the confirmation back from the receiver.

Andrew has experienced this with two different lite users.

**Comment by uncleJim21** [comment]
One proposal I had was to use checksums on the messageIds array. That way the client and server can easily "ping" each other with a value to determine if a resync is needed with relatively low overhead.

**Comment by uncleJim21** [comment]
Here's a very simple prototype I made. The idea is that you could quickly and easily detect a discrepancy in the checksum between the local and remote database and take error correction action if the checksums are not identical:
<img width="1348" alt="image" src="https://github.com/stakwork/sphinx-relay/assets/96802642/c544d3c5-8bc4-402c-ba5a-11efc2847ebf">

This technique is pretty common in communication systems due to its effectiveness: you only need that one string to determine if there's an error so you can check for errors quite frequently. One way I propose we do it is just have the client & relay compare checksums every N seconds while inside a chat and if there's a discrepancy, initiate a "self healing" routine where it resends any messages on local that are not recognized as sent by the relay.

**Comment by Evanfeenstra** [comment]
Nice idea... the bigger problem though is that messages might never make it to your relay because of lightning liquidity issues, or MQTT issues

**Comment by uncleJim21** [comment]
Just so we're clear the chain of custody looks something like

(sender client) --(1)---> (sender relay) ---(2)---> (recipient relay) ----(3)---->(recipient client)

It seems you're mostly concerned about step 2 which seems to be the problem here?

**Comment by Evanfeenstra** [comment]
Right. Latest relay master waits for step 2 before returning a success or error message to the sender, so that should help a lot with "silent" failed msgs

**Comment by uncleJim21** [comment]
So are these current failures silent because the relay isn't update to latest master?

One thought for how we can help on the client side is to detect a pattern of frequent failures and make recommendations such as adding more liquidity and alert the user?
